### PR TITLE
FOCUS #145: List Unit Price spec changes and supporting content

### PR DESCRIPTION
**Pull request by ijurica**

### CODE REVIEW

File: supporting_content/dimensions/listunitprice.md
Comment (flanakin) [inline supporting_content/dimensions/listunitprice.md:11]: The difference between the pricesheet and cost details is intentional and should be the difference between the pricing unit and usage unit. I don't think that's worth calling out in this table. It's fine to call it out in supporting content, but we should add context about why it's like that.

File: supporting_content/dimensions/listunitprice.md
Comment (flanakin) [inline supporting_content/dimensions/listunitprice.md:20]: What does `UCs` stand for? We should probably spell that out.

File: supporting_content/dimensions/listunitprice.md
Comment (flanakin) [inline supporting_content/dimensions/listunitprice.md:22]: We should either move the data to this repo or open this spreadsheet for anonymous access. Doesn't make sense to link to a doc as part of a spec that nobody has access to.

**Review by flanakin — APPROVED** [review]

File: specification/dimensions/listunitprice.md
Comment (ijurica) [reply specification/dimensions/listunitprice.md:25]: One of the members pointed out that **null** is probably more appropriate than '0' in certain cases (e.g. for chargeType: **tax**). As a result, we made the change to 'false' and underscored that '... MUST NOT contain null values in the case of usage-based charges.' 

**QUESTION**: Should we enforce this for both **usage-based and purchase-based** charges? 
_...MUST NOT contain null values for both usage-based and purchase-based charges_

File: supporting_content/dimensions/listunitprice.md
Comment (ijurica) [reply supporting_content/dimensions/listunitprice.md:11]: **ASSUMPTIONS:**

I assumed that both Retail Prices API and Cost details provide PricingUnit and ListUnitPrice (per PricingUnit, not UsageUnit)

- Retail Prices API via retailPrice, unitOfMeasure
- Cost details via PayGPrice, unitOfMeasure

I anticipated alignment between the values but observed certain discrepancies and wasn't sure about the cause.


**SAMPLE DATA:**

- Retail Prices API response:

```
{
	"currencyCode": "USD",
	...
	"retailPrice": 0.00036,
	...
	"meterId": "b9e5e77c-a0b3-4a2c-9b8b-57fa54f31c52",
	...
	"unitOfMeasure": "10K",
	...
}
```

- Sample PAYG Cost details record
  - meterId: b9e5e77c-a0b3-4a2c-9b8b-57fa54f31c52
  - unitOfMeasure: 10K
  - PayGPrice: 0.00036

- Sample CSP Cost details record
  - meterId: b9e5e77c-a0b3-4a2c-9b8b-57fa54f31c5
  - unitOfMeasure: 10K
  - PayGPrice: 0.0003707


**QUESTION:** 

If the assumption regarding the Azure Cost Details PayGPrice is incorrect, **would it be acceptable to exclude the reference to PayGPrice and rephrase it as: _'...ListUnitPrice is not available in Cost Details, but can be obtained through the Retail Prices API...'_?**

File: supporting_content/dimensions/listunitprice.md
Comment (ijurica) [inline supporting_content/dimensions/listunitprice.md:9]: Correction required: The reference should be to 'pricing/publicOnDemandRate' instead of 'pricing/publicOnDemandCost'.

File: specification/dimensions/listunitprice.md
Comment (ijurica) [reply specification/dimensions/listunitprice.md]: Suggested change applied

File: supporting_content/dimensions/listunitprice.md
Comment (nfonrose) [inline supporting_content/dimensions/listunitprice.md:112]:
```suggestion
In both sybtypes, the count of charged units resets at the beginning of each interval.
```
```suggestion
In both subtypes, the count of charged units resets at the beginning of each interval.
```

File: supporting_content/dimensions/listunitprice.md
Comment (nfonrose) [inline supporting_content/dimensions/listunitprice.md:10]: Aren't we missing the "Google Cloud Pricing API" here ?
 - It's functionally equivalent to the "AWS Price List API"
 - It contains a "list price" value
 - It also has tiering concepts
https://cloud.google.com/billing/docs/how-to/get-pricing-information-api

Extract from the page: The following example shows the pricing details for the SKU ID 0012-B7F2-DD14.
```
{
  "name": "skus/0012-B7F2-DD14/price",
  "currencyCode": "USD",
  "valueType": "rate",
  "rate": {
    "tiers": [
      {
        "startAmount": {
          "value": "0"
        },
        "listPrice": {
          "currencyCode": "USD",
          "nanos": 1397000
        }
      }
    ],
    "unitInfo": {
      "unit": "GiBy.h",
      "unitDescription": "gibibyte hour",
      "unitQuantity": {
        "value": "1"
      }
    },
    "aggregationInfo": {
      "level": "ACCOUNT",
      "interval": "MONTHLY"
    }
  }
}
```

**Review by nfonrose — CHANGES_REQUESTED** [review]

File: supporting_content/dimensions/listunitprice.md
Comment (ijurica) [reply supporting_content/dimensions/listunitprice.md]: Provided more information about flat-rate and usage-dependent pricing

File: specification/dimensions/listunitprice.md
Comment (udam-f2) [reply specification/dimensions/listunitprice.md:25]: tend to agree. If we put null, aggregations won't be wrong. If we put 0 artificially, then we may screw up some numbers and make it unclear when the actual rate is 0. Whatever we decide, we need to be consistent with the other columns as well. 

The alternative is if the cost of tax is $100, we say quantity is 1 and unit is 'count' and rate is '100' type thing which could work though that rate is always moving and doesn't really mean anything.

File: supporting_content/dimensions/listunitprice.md
Comment (ijurica) [reply supporting_content/dimensions/listunitprice.md]: Crucial content has been copied to MD files. 

**Notes:** 

- The link to one of the docs is still present for additional information, but since the main content crucial for ListUnitPrice in the billing context is available within the supporting materials, **we can omit the link.** 
- We should decide **how to handle any significant spreadsheets** when/if we decide to prepare an **appendix for Pricing support scenarios**.

File: supporting_content/dimensions/listunitprice.md
Comment (ijurica) [reply supporting_content/dimensions/listunitprice.md]: Forgot to mention the API. Included both in Providers mapping and sample UC scenarios related to alternative data sources.

File: specification/dimensions/listunitprice.md
Comment (ijurica) [reply specification/dimensions/listunitprice.md:26]: Since it's defined as the provider-published price, I prefer null in the case of tax-type charges. Still not sure if null should be allowed in case of purchase-based charges.

File: specification/dimensions/listunitprice.md
Comment (ijurica) [reply specification/dimensions/listunitprice.md]: Applied suggested changes and adjusted the Content Constraint as well.

File: specification/dimensions/listunitprice.md
Comment (ijurica) [reply specification/dimensions/listunitprice.md]: The issue discussed and addressed in another conversation.

File: specification/dimensions/listunitprice.md
Comment (udam-f2) [inline specification/dimensions/listunitprice.md:5]: Looks like BilledCost related sentence got in some how. I assume this was put so we can make the line below consistent with the other 'Metric' columns.

File: specification/dimensions/listunitprice.md
Comment (udam-f2) [inline specification/dimensions/listunitprice.md:6]:
```suggestion
The ListUnitPrice column MUST be present in the billing data. This column MUST be a numeric value of type Decimal within the range of non-negative decimal values. It MUST NOT contain null values in cases where the [ChargeType](#chargetype) is 'usage' or 'purchase'.
```

File: specification/dimensions/listunitprice.md
Comment (udam-f2) [inline specification/dimensions/listunitprice.md:18]:
```suggestion
Represents the suggested provider published price for a single pricing measurement unit, which incorporates volume/tier-based rates while excluding any negotiated or commitment-based discounts.
```

File: specification/dimensions/listunitprice.md
Comment (udam-f2) [inline specification/dimensions/listunitprice.md:3]:
```suggestion
The List Unit Price represents the suggested provider published price for a single pricing measurement unit, which incorporates volume/tier-based rates while excluding any negotiated or commitment-based discounts. This price is denominated in the [Billing Currency](#billingcurrency). It is used to calculate List Cost and savings based on rate optimization activities.
```

**Review by udam-f2 — APPROVED** [review]

File: specification/dimensions/listunitprice.md
Comment (ijurica) [reply specification/dimensions/listunitprice.md]: Introduced by mistake :(. Deleted it.

File: supporting_content/dimensions/listunitprice.md
Comment (ijurica) [reply supporting_content/dimensions/listunitprice.md]: Discussed at maintainers' meeting. For the time being, links to spreadsheets within the supporting content are acceptable, even if those are not available to the public.

File: supporting_content/dimensions/listunitprice.md
Comment (ijurica) [reply supporting_content/dimensions/listunitprice.md]: Discussed and decided to keep this note about discrepancy with PayGPrice in the supporting content.

File: supporting_content/appendix/quantity_driven_pricing_strategies.md
Comment (udam-f2) [inline supporting_content/appendix/quantity_driven_pricing_strategies.md:20]:
```suggestion
* **tier-based pricing:** In tier-based pricing, unit prices change as the number of charged units increases. You are charged for the consumed quantity within each tier based on the associated rate for each tier.
```

File: supporting_content/appendix/quantity_driven_pricing_strategies.md
Comment (udam-f2) [inline supporting_content/appendix/quantity_driven_pricing_strategies.md:1]: Is this meant to be in supporting content?

File: supporting_content/appendix/quantity_driven_pricing_strategies.md
Comment (ijurica) [reply supporting_content/appendix/quantity_driven_pricing_strategies.md:19]: It only makes sense if it's synchronized with the billing period? Would you agree?
I'll replace 'interval' with 'billing period' for now, and later we can revisit and discuss further.

File: supporting_content/appendix/quantity_driven_pricing_strategies.md
Comment (ijurica) [reply supporting_content/appendix/quantity_driven_pricing_strategies.md:20]: I'll just replace 'rate' with 'unit price'.

File: supporting_content/appendix/quantity_driven_pricing_strategies.md
Comment (ijurica) [reply supporting_content/appendix/quantity_driven_pricing_strategies.md]: Replaced all mentions of 'interval' with 'billing period.' 
Resolving this conversation as there is a pending discussion on the quantity-driven pricing strategies topic in general.

File: supporting_content/metrics/listunitprice.md
Comment (timdcook123) [inline supporting_content/metrics/listunitprice.md:67]: A price refers to a fixed amount of money. A rate refers to a set amount of money which is charged according to some period of time or some number of items. The price is $50 vs. The rate is $50/hour.

**Review by ljadvey — APPROVED** [review]

**Pull request merged**
